paddle/platform/device_context.h

```diff
@@ -74,8 +75,7 @@ class CUDADeviceContext : public DeviceContext {
  public:
   explicit CUDADeviceContext(const GPUPlace gpu_place) : gpu_place_(gpu_place) {
     GPUPlaceGuard guard(gpu_place_);
-    paddle::platform::throw_on_error(cudaStreamCreate(&stream_),
-                                     "cudaStreamCreate failed");
+    PADDLE_ENFORCE(cudaStreamCreate(&stream_), "cudaStreamCreate failed");
```

Comment:
Yeah, @wangkuiyi 

```
PADDLE_ENFORCE(cudaStreamCreate(&stream_) == CUDA_SUCCESS);
```
 will also impede us to print apposite error info for cuda, cudnn, cublas, curand.